test/sanity/src/context.ts

```diff
@@ -0,0 +1,429 @@
+		if (!response.ok) {
+			this.error(`Failed to fetch ${url}: ${response.status} ${response.statusText}`);
+		}
+
+		if (response.body === null) {
+			this.error(`Response body is null for ${url}`);
+		}
+
+		return response as Response & { body: NodeJS.ReadableStream };
+	}
+
+	/**
+	 * Fetches metadata for a specific VS Code release target.
+	 * @param target The target platform (e.g., 'cli-linux-x64').
+	 * @returns The target metadata.
+	 */
+	public async fetchMetadata(target: string): Promise<ITargetMetadata> {
+		const url = `https://update.code.visualstudio.com/api/versions/commit:${this.commit}/${target}/${this.quality}`;
+
+		this.log(`Fetching metadata for ${target} from ${url}`);
+		const response = await this.fetchNoErrors(url);
+
+		const result = await response.json() as ITargetMetadata;
+		if (result.url === undefined || result.sha256hash === undefined) {
+			this.error(`Invalid metadata response for ${target}: ${JSON.stringify(result)}`);
+		}
+
+		this.log(`Fetched metadata for ${target}: ${JSON.stringify(result)}`);
+		return result;
+	}
+
+	/**
+	 * Downloads installer for specified VS Code release target.
+	 * @param target The target platform (e.g., 'cli-linux-x64').
+	 * @returns The path to the downloaded file.
+	 */
+	public async downloadTarget(target: string): Promise<string> {
+		const { url, sha256hash } = await this.fetchMetadata(target);
+		const filePath = path.join(this.createTempDir(), path.basename(url));
+
+		this.log(`Downloading ${url} to ${filePath}`);
+		const { body } = await this.fetchNoErrors(url);
+
+		const stream = fs.createWriteStream(filePath);
+		await new Promise<void>((resolve, reject) => {
+			body.on('error', reject);
+			stream.on('error', reject);
+			stream.on('finish', resolve);
+			body.pipe(stream);
+		});
+
+		this.log(`Downloaded ${url} to ${filePath}`);
+		this.validateSha256Hash(filePath, sha256hash);
+
+		if (TestContext.authenticodeInclude.test(filePath) && os.platform() === 'win32') {
+			this.validateSignature(filePath);
+		}
+
+		return filePath;
+	}
+
+	/**
+	 * Validates the SHA-256 hash of a file.
+	 * @param filePath The path to the file to validate.
+	 * @param expectedHash The expected SHA-256 hash in hexadecimal format.
+	 */
+	public validateSha256Hash(filePath: string, expectedHash: string) {
+		this.log(`Validating SHA256 hash for ${filePath}`);
+
+		const buffer = fs.readFileSync(filePath);
+		const hash = createHash('sha256').update(buffer).digest('hex');
+
+		if (hash !== expectedHash) {
+			this.error(`Hash mismatch for ${filePath}: expected ${expectedHash}, got ${hash}`);
+		}
+	}
+
+	/**
+	 * Validates the Authenticode signature of a Windows executable.
+	 * @param filePath The path to the file to validate.
+	 */
+	public validateSignature(filePath: string) {
+		this.log(`Validating Authenticode signature for ${filePath}`);
+
+		const result = this.run('powershell', '-Command', `Get-AuthenticodeSignature "${filePath}" | Select-Object -ExpandProperty Status`);
```

Comment:
I'm guessing this is where my errors come from. We can't assume Powershell is available everywhere. Or is this meant to be run only on Windows platforms?